### ISSUE TITLE
Enhancement: [no-floating-promises] foreach of async callbacks

### ISSUE THREAD

**Issue by soryy708** [typescript-eslint/typescript-eslint#9245]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

`no-floating-promises` should check for `.forEach`es that receive an `async` callback (or callback that returns a promise)

### Fail

```TypeScript
[1, 2, 3].forEach(async (x) => x + 1);
```

### Pass

```TypeScript
await Promise.all([1, 2, 3].map(async (x) => x + 1));
```

### Additional Info

If you `forEach` as shown, the promises returned by the callbacks are floating.
